File: Xamarin.iOS/xamarinMeetup2_x_ios/SecondViewController.cs
using System;
using System.Threading.Tasks;
using UIKit;
using xamarinMeetup2;

namespace xamarinMeetup2_x_ios
{
    public partial class SecondViewController : UIViewController
    {
        MathModel mathModel;

        protected SecondViewController(IntPtr handle) : base(handle)
        {
            // Note: this .ctor should not contain any initialization logic.
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();
            // Perform any additional setup after loading the view, typically from a nib.
            mathModel = new MathModel();
        }

        public override void DidReceiveMemoryWarning()
        {
            base.DidReceiveMemoryWarning();
            // Release any cached data, images, etc that aren't in use.
        }

        partial void CalculateButton_TouchUpInside(UIButton sender)
        {
            Task.Run(() =>
                    {
                        // calculate in background
                        var arithmetic = "Arithmetic: " + mathModel?.ArithmeticPerformance;
                        var collections = "Collections: " + mathModel?.CollectionPerformance;
                        var strings = "Strings: " + mathModel?.StringPerformance;

                        // update in UI thread
                        InvokeOnMainThread(() =>
                                {
                                    arithmeticLabel.Text = arithmetic;
                                    collectionsLabel.Text = collections;
                                    stringsLabel.Text = strings;
                                }
                            );


                    });
        }
    }
}

File: Native.iOS/xm2 Native/SecondViewController.swift
//
//  SecondViewController.swift
//  xm2 Native
//
//  Created by Yury Nevalenny on 13/11/2016.
//  Copyright © 2016 Rahmobile. All rights reserved.
//

import UIKit

class SecondViewController: UIViewController {
    @IBOutlet weak var arithmeticLabel: UILabel!
    @IBOutlet weak var collectionsLabel: UILabel!
    @IBOutlet weak var stringsLabel: UILabel!
    @IBOutlet weak var calculateButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func Calculate(_ sender: Any) {
        DispatchQueue.global(qos: .background).async {
            let mathModel = MathModel()
            mathModel.calculate()
            
            DispatchQueue.main.async {
                self.arithmeticLabel.text = "Arithmetic: "+mathModel.arithmeticPerformance
                self.collectionsLabel.text = "Collections: "+mathModel.collectionsPerformance
                self.stringsLabel.text = "Strings: "+mathModel.stringsPerformance
            }
        }
    }
}

